### DISCUSSION TITLE
Blazor .NET 8 - Enhanced Form Navigation

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#85] [General]
# Blazor .NET 8 - Enhanced Form Navigation

There are many new cool features with .NET 8 and Blazor in particular. In this blog post, I want to highlight a feature that I believe is very useful in the new context Blazor is living.

https://steven-giesel.com/blogPost/2e87048d-cbf3-4378-8938-efd4648453a1

**Comment by dimohy** [comment]
Thanks for this quick overview of the new features available in .NET 8 Blazor.

I know that in .NET 8, Blazor can mix Server and WebAssembly interactions, but do you know how to keep state values mutual when doing so?

**Reply by linkdotnet** [reply]
Hey. I am not 100% certain if I understand your concrete problem - can you elaborate a bit more?

General speaking, while you can mix WebAssembly and Server, you can't mix them inside a given tree.
```no-class
          Root
    (Static Serverside Rendering)
         /      \
        /        \
  Child 1      Child 3
(InteractiveServer) (InteractiveWebAssembly)
     |
     |
  Child 2

```

Child 2 can not be `InteractiveWebAssembly` in that case. Once you switched to a specific mode, you have to use the mode in the hierarchy tree.

**Reply by linkdotnet** [reply]
But my guess is that you are talking about having a state that is shared between them!?

**Reply by linkdotnet** [reply]
If you are talking about the latter one: https://github.com/dotnet/aspnetcore/issues/47796

So basically, there is nothing out of the box helping you.
